### PR TITLE
Fix excessive compilation times due to speculative parsing after an incomplete string

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/72557

Don't love this approach, but couldn't think of anythign better.  See comments in bug for what's going on.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@RikkiGibson @cston ptal

File: src/Compilers/CSharp/Portable/Parser/LanguageParser.cs [hunk 1]
```diff
@@ -13348,6 +13363,7 @@ private bool IsInQuery
         /// All the callbacks should passed as static lambdas or static methods to prevent unnecessary delegate
         /// allocations.
         /// </remarks>
+
```
Comment (333fred) [inline src/Compilers/CSharp/Portable/Parser/LanguageParser.cs:13366]: Stray newline?

File: src/Compilers/CSharp/Portable/Parser/LanguageParser.cs [hunk 2]
```diff
@@ -13357,6 +13373,31 @@ private bool IsInQuery
...
+                isPossibleElement,
+                parseElement,
+                immediatelyAbort: null,
+                skipBadTokens,
+                allowTrailingSeparator,
+                requireOneElement,
+                allowSemicolonAsSeparator);
+        }
+
+#nullable enable
```
Comment (333fred) [inline src/Compilers/CSharp/Portable/Parser/LanguageParser.cs:13389]: Why not enable the above method?

File: src/Compilers/CSharp/Portable/Parser/LanguageParser.cs [hunk 3]
```diff
@@ -1058,6 +1059,20 @@ private AttributeSyntax ParseAttribute()
                     static (p, closeKind) => p.CurrentToken.Kind == closeKind,
                     expectedKind, closeKind);
             }
+
+            static bool immediatelyAbort(AttributeArgumentSyntax argument)
+            {
+                // We can be very thrown off by incomplete strings in an attribute argument (especially as the lexer
+                // will restart on the next line with the contents of the string then being interpreted as more
+                // arguments).  Bail out in this case to prevent going off the rails.
+                if (argument.expression is LiteralExpressionSyntax { Kind: SyntaxKind.StringLiteralExpression, Token: var literalToken } &&
```
Comment (333fred) [inline src/Compilers/CSharp/Portable/Parser/LanguageParser.cs:1068]: What about interpolated strings with newlines in them?

File: src/Compilers/CSharp/Portable/Parser/LanguageParser.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Portable/Parser/LanguageParser.cs:13389]: this was hte method where i added something nullable :)

File: src/Compilers/CSharp/Portable/Parser/LanguageParser.cs (hunk 3 above)
Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Portable/Parser/LanguageParser.cs:1068]: sure.  i can extend to that case as well.

File: src/Compilers/CSharp/Portable/Parser/LanguageParser.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Portable/Parser/LanguageParser.cs:13366]: yup!

**Review by 333fred — APPROVED** [review]

**Comment by RikkiGibson** [comment]
It looks like some of the Helix runs may be timing out.

**Review by cston — APPROVED** [review]

**Pull request merged**